internal/core/application/operator_service.go

```diff
@@ -625,19 +620,30 @@ func (o *operatorService) GetCollectedMarketFee(
 		return nil, err
 	}
 
-	fees := make([]Fee, 0)
+	fees := make([]FeeInfo, 0, len(trades))
 	total := make(map[string]int64)
-	for _, v := range trades {
-		fees = append(fees, Fee{
-			FeeAsset:   v.MarketFeeAsset,
-			BasisPoint: v.MarketFee,
-		})
-
-		if val, ok := total[v.MarketFeeAsset]; ok {
-			total[v.MarketFeeAsset] = val + v.MarketFee
+	for _, trade := range trades {
+		feeBasisPoint := trade.MarketFee
+		swapRequest := trade.SwapRequestMessage()
+		feeAsset := swapRequest.GetAssetP()
+		amountP := swapRequest.GetAmountP()
+		_, feeAmount := mathutil.LessFee(amountP, uint64(feeBasisPoint))
+		var marketPrice decimal.Decimal
```

Comment:
we may want to skip the else here, assing the quotePrice by default, if not then assing the base one.

Reply:
Done.